### PR TITLE
Minor agent updates

### LINKED ISSUE #8085: [Bug]: Sources not returned by OpenAI Agent query methods in cases where tools are also OpenAI Agents
### Bug Description

Response returned by `BaseAgent` in its `query` methods does not include `source_nodes` in the object returned. This results in `source_nodes` not being populated correctly when Agents are also used as `QueryEngineTools`

### Version

0.8.43.post1

### Steps to Reproduce

- Create multiple VectorStore Indices, and corresponding query engines. 
- Create query engine tools for each engine, and make OpenAIAgents (sub-agents) with those tools. 
- Create a "master" OpenAI Agent with tools, where the tools are OpenAIAgents from the previous step wrapped in QueryEngineTools (query engines for these tools are the sub-agents). 
- Use the "master" OpenAIAgent to answer a query, and print response.sources or response.source_nodes, which should yield empty lists. 

### Relevant Logs/Tracbacks

_No response_

### PR DESCRIPTION
**Pull request by vishhvak**
# Description

This PR includes changes that fix issue #8085 and implement feature #8084. The changes involve modifications to the `BaseAgent` and `ReActAgent` classes to handle `sources` correctly. `ReActAgent` now returns `sources` with its response. Agents now correctly return source_nodes when they are used as `QueryEngineTool`s using their `query` and `aquery` methods from the `BaseAgent` class.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

Fixes #8085, Implements #8084

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by vishhvak** [comment]
@logan-markewich I raised a PR for the agent sources fix we discussed a week or two back :) Hope this is okay, please do let me know if I've missed anything or have made any mistakes! My first few times doing open source, so I'm still quite the noob :)

File: llama_index/agent/openai_agent.py [hunk 1]
```diff
@@ -236,7 +240,10 @@ async def _acall_function(self, tools: List[BaseTool], function_call: dict) -> N
                 tools, function_call, verbose=self._verbose
             )
             event.on_end(payload={EventPayload.FUNCTION_OUTPUT: str(tool_output)})
-        self.sources.append(tool_output)
+        if isinstance(tool_output, AgentChatResponse):
```
Comment (logan-markewich) [inline llama_index/agent/openai_agent.py:243]: Isn't the tool output always a `ToolOutput` object, i.e. Line 54

In what case would this not be true? 👀

Comment (vishhvak) [reply llama_index/agent/openai_agent.py]: @logan-markewich Sorry for this! No idea what I was thinking, was just plain stupid. Fixed it, the correction to `BaseAgent` query methods suffices.

**Comment by vishhvak** [comment]
@logan-markewich updated the commits and fixed issues. Can you let me know if this are okay, or if I need to make any other changes? :)

**Pull request merged**